packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -222,11 +262,26 @@ class LinkViewController extends PlatformViewController {
   /// `click` from the browser.
   static void registerHitTest(LinkViewController controller) {
     _hitTestedViewId = controller.viewId;
+    followLinkTriggered = true;
+
+    if (triggeredMouseEvent != null) {
```

Comment:
We should do the same for keyboard events (we probably don't need a `triggeredKeyboardEvent`, but we need a boolean at least).

Here's how I would do it, but feel free to adjust/modify it to how you see fit:
```dart
// Set to true in registerHitTest()
static bool followLinkTriggered = false;

// Set to true in _onGlobalClick() and _onGlobalKeydown()
static bool domEventTriggered = false;

// Set only in _onGlobalClick()
static html.Event? triggeredEvent;
```

Reply:
Based on my testing, the keyboard event path doesn't deviate from hit test -> global keydown.  Do you think this boolean is necessary because there would be a case where the browser keydown handler executes before the framework handles it?

I had to be pretty careful about this in the click handler because we were running into issues where the dom event trigger was being set to true for arbitrary clicks (e.g. clicking elsewhere on the app).  This led to issues where we would trigger the link twice if the event path went hit test -> global click